astropy/units/decorators.py

```diff
@@ -115,16 +116,38 @@ def wrapper(*func_args, **func_kwargs):
                 # If the target unit is empty, then no unit was specified so we
                 # move past it
                 if target_unit is not funcsigs.Parameter.empty:
+
+                    if isinstance(target_unit, str):
+                        from .core import Unit
+
+                        try: # unit passed in as a string
+                            target_unit = Unit(target_unit)
+                            phys_type = False
+                        except ValueError:
+                            phys_type = True
+
+                        if phys_type:
+                            # user specified a physical type instead of a unit
+                            ureg = get_current_unit_registry()
+                            try:
+                                physical_type_id = _unit_physical_mapping[target_unit]
+                            except KeyError:
+                                raise ValueError("Invalid physical type '{0}'."
+                                                 .format(target_unit))
+                            target_units = ureg._by_physical_type[physical_type_id]
+                            target_unit = target_units.pop() # just grab the first valid unit
+
                     try:
                         equivalent = arg.unit.is_equivalent(target_unit,
                                                   equivalencies=self.equivalencies)
 
                         if not equivalent:
                             raise UnitsError("Argument '{0}' to function '{1}'"
                                              " must be in units convertible to"
-                                             " '{2}'.".format(param.name,
+                                             " physical type '{2}'."
+                                             .format(param.name,
                                                      wrapped_function.__name__,
-                                                     target_unit.to_string()))
+                                                     target_unit.physical_type))
```

Comment:
Ah, though, I just realized that I would then need a better way to get out the target unit from the physical type. If you look at what is now L138, I just use `.pop()` to grab *any* unit that matches the physical type. But that will change at random, and it is often a strange prefixed unit. Any ideas there? I want to get out something like the "preferred representation" for a given physical type, given all enabled units...